app/lib/puppet_fact_parser_extensions.rb

```diff
@@ -2,42 +2,28 @@ module PuppetFactParserExtensions
   extend ActiveSupport::Concern
 
   included do
-    # In Foreman 1.8 these two methods have been removed, we could reuse discovery_mac_fact_name in suggested_primary_interface(host) if needed
-    if instance_methods.include?(:primary_interface)
-      alias_method_chain :primary_interface, :discovery_fact
-    end
-
-    if instance_methods.include?(:ip)
-      alias_method_chain :ip, :discovery_fact
+    begin
+      raise "method: suggested_primary_interface" unless instance_methods.include?(:suggested_primary_interface)
+      raise "method: parse_interfaces?" unless instance_methods.include?(:parse_interfaces?)
```

Comment:
Our tests did not catch it apparently and current tests will unlikely catch this too, because there are no mocha expectations at the moment. We don't need to stub anything for this particular unit test.

If you insist, I can remove this and add two extra tests with an expectation. I just feel this is much more transparent. Someone else can disable or remove the test thinking that it is correct and if that passes review...